src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -308,11 +308,29 @@ private ProcessPriorityClass PriorityClassCore
         }
 
         /// <summary>Checks whether the argument is a direct child of this process.</summary>
-        private bool IsParentOf(Process possibleChildProcess) =>
-            Id == possibleChildProcess.ParentProcessId;
+        private bool IsParentOf(Process possibleChildProcess)
+        {
+            try
+            {
+                return Id == possibleChildProcess.ParentProcessId;
+            }
+            catch (Exception e) when (IsProcessInvalidException(e))
+            {
+                return false;
+            }
+        }
 
-        private bool Equals(Process process) =>
-            Id == process.Id;
+        private bool Equals(Process process)
```

Comment:
Are `Equals` and `IsParentOf`  methods called only from `Process.NonUap.cs`? (if not adding the catch blocks could have side effects for other callers)

Reply:
It's also called in one place from Process.Win32.cs, but it was eating the same exceptions:
https://github.com/dotnet/runtime/pull/50496/files#diff-7df572ea49f2a0e61116ce269d48b0b828274216cfdd7053750f5490063cfc54L392